doc/website/blog/2022-03-14-ent-hooks-examples.md

```diff
@@ -0,0 +1,434 @@
+
+### Keeping our secrets safe:
+
+Many times, we need to store strings that include secrets, a good example can be a connection
+string to a database, usually in the form of ```mysql://root:pass@localhost:3306)```, here, for example, it’s very
+easy to see  that we don't want the password to unintentionally be retrieved by a query unless required by the server
+(for example to connect to a database).
+The required behavior is best described by this test:
+```go
+func TestUserConnectionStringHook(t *testing.T) {
+	ctx := context.Background()
+	c := enttest.Open(t, dialect.SQLite,
+		"file:TestSchemaConfHooks?mode=memory&cache=shared&_fk=1",
+	)
+	u := c.User.Create().SetName("Yoni").SetConnectionString("mysql://root:pass@localhost:3306)").SaveX(ctx)
+	require.Equal(t, "mysql://root:****@localhost:3306)", u.ConnectionString)
+	require.Equal(t, "pass", u.Password)
+	require.Equal(t, "mysql://root:pass@localhost:3306)", u.FullConnectionString())
+}
+```
+
+As you can see, inserting a connection string updates the string to have `****` instead of the password and stores the 
+password in the ```Password``` field that is defined as [sensitive](https://entgo.io/docs/schema-fields#sensitive-fields).
+To have  the connection string in full valid form we created a utility function ```FullConnectionString()```.
+
+#### Recipe:
+
+First thing, add the hook to your schema:
+```go
+// Hooks of the User.
+func (User) Hooks() []ent.Hook {
+	return []ent.Hook{
+		hook.If(clearConnectionString,
+			hook.And(hook.Or(hook.HasOp(ent.OpUpdateOne), hook.HasOp(ent.OpCreate)), hook.HasFields(user.FieldConnectionString)),
+		),
+	}
+}
+```
+* Do not forget to run go generate or the hook will not be registered.
+
+A few hook helpers are used here, ```clearConnectionString``` is the name of the hook function, and it will run if a user 
+is Updated or Created and the mutation has the field ```connection_string``` (if we change only the name of the user there is no need to do anything right?).
+
+The hook:
+```go
+func clearConnectionString(next ent.Mutator) ent.Mutator {
+	return hook.UserFunc(func(ctx context.Context, m *gen.UserMutation) (ent.Value, error) {
+		cs, ok := m.ConnectionString()
+		if !ok {
+			return next.Mutate(ctx, m)
+		}
+		sp := strings.Split(cs, "@")
+		if len(sp) != 2 {
+			return next.Mutate(ctx, m)
+		}
+		sp = strings.Split(sp[0], ":")
+		if len(sp) != 3 {
+			return next.Mutate(ctx, m)
+		}
+		pass := sp[2]
+		m.SetPassword(pass)
+		m.SetConnectionString(strings.ReplaceAll(cs, pass, "****"))
+		return next.Mutate(ctx, m)
+	})
+} 
+```
+We validate that the mutation has a connection string (if not just let the mutation continue).
+A bit of string acrobatics, and we have the password extracted from the string, we store it in another field - the ```password``` and
+change the ```connection_string``` with a masked version. 
+
+To get the full connection string we add a utility function 
+```go
```

Comment:
Again, add the title so the user knows that this is an extra file and not `user.go`